### PR TITLE
tools: add doc generate command

### PR DESCRIPTION
**Pull request by amyXia1994**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [X] N
- [ ] Y 

#### 2. What is the scope of this PR (e.g. component or file name):

cmds/kcl-go/command/cmd_doc.go
pkg/tools/doc/doc_test.go
pkg/tools/doc/kdoc.go
pkg/tools/doc/templates/schema.gotmpl
go.mod
go.sum

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

tools: add doc generate command

- the `doc` subcommand is now set to `Hidden` and will be turned on when the feature is ready to be released
- this PR adds the `kcl-go doc generate` command and implements the process from cli input to render markdown document to the target output directory. The more complete functionalities(such as starting a local documentation server, HTML/openapi format output, and the document template refine, ...) will be improved later.

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [X] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [X] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

### CODE REVIEW

File: pkg/tools/doc/kdoc.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+func init() {
+	var err error
+	// todo: change to nested template files
+	tmpl, err = template.New("doc.md").Funcs(funcMap()).Parse(schemaDocTmpl)
+	if err != nil {
+		panic(err)
+	}
+}
+
+type GenContext struct {
```
Comment (Peefy) [inline pkg/tools/doc/kdoc.go:30]: Need more documents for public structures and functions.

File: pkg/tools/doc/kdoc.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+const (
+	MD   = "md"
+	HTML = "html"
+)
+
+type Format int
+
+const (
+	Html Format = iota
+	Markdown
```
Comment (Peefy) [inline pkg/tools/doc/kdoc.go]: Using MD and Markdown as variable names simultaneously may cause confusion.

File: cmds/kcl-go/command/cmd_doc.go [hunk 3]
```diff
@@ -4,29 +4,88 @@ package command
...
+
 func NewDocCmd() *cli.Command {
 	return &cli.Command{
-		Hidden:          false,
-		SkipFlagParsing: true,
-		Name:            "doc",
-		Usage:           "show documentation for package or symbol",
+		Hidden: true,
+		Name:   "doc",
+		Usage:  "show documentation for package or symbol",
```
Comment (Peefy) [inline cmds/kcl-go/command/cmd_doc.go:17]: U can add some examples in the kcl doc usage docs.

File: cmds/kcl-go/command/cmd_doc.go [hunk 4]
```diff
@@ -4,29 +4,88 @@ package command
...
+		Subcommands: []*cli.Command{
+			{
+				Name:  "generate",
+				Usage: "generates documents from code and examples",
```
Comment (Peefy) [inline cmds/kcl-go/command/cmd_doc.go:26]: U can add some examples in the kcl doc usage docs. e.g. `kcl-go doc generate /path/to/file` or `kcl-go doc generate /path/to/dir`

File: pkg/tools/doc/kdoc.go (hunk 1 above)
Comment (amyXia1994) [reply pkg/tools/doc/kdoc.go:30]: fixed

File: pkg/tools/doc/kdoc.go (hunk 2 above)
Comment (amyXia1994) [reply pkg/tools/doc/kdoc.go]: fixed

File: cmds/kcl-go/command/cmd_doc.go (hunk 3 above)
Comment (amyXia1994) [reply cmds/kcl-go/command/cmd_doc.go:17]: done~

File: cmds/kcl-go/command/cmd_doc.go (hunk 4 above)
Comment (amyXia1994) [reply cmds/kcl-go/command/cmd_doc.go:26]: done~

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**